### PR TITLE
run some pre-commit checks

### PR DESCRIPTION
**Pull request by blackliner**

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -86,7 +86,7 @@ markdown_extensions:
       custom_fences:
         - name: mermaid
           class: mermaid
-          format: !!python/name:pymdownx.superfences.fence_code_format
+          format: !%21python/name:pymdownx.superfences.fence_code_format
```
Comment (blackliner) [inline mkdocs.yml:89]: mostly unsure about this change 🤔

File: docs/vlab/demo.md [hunk 2]
```diff
@@ -7,7 +7,7 @@ You can find instructions on how to setup VLAB in the [Overview](overview.md) an
 
 ## Default topology
 
-The default topology is Spine-Leaf with 2 spines, 2 MCLAG leafs and 1 non-MCLAG leaf.
+The default topology is Spine-Leaf with 2 spines, 2 MCLAG leaves and 1 non-MCLAG leaf.
```
Comment (blackliner) [inline docs/vlab/demo.md:10]: unwanted change

File: docs/release-notes/index.md [hunk 3]
```diff
@@ -67,7 +67,7 @@ Switches not directly connecting to the controllers can chain boot via the data
 
 CLOS/LEAF-SPINE and Collapsed Core topologies are supported.
 
-#### LEAF Roles for CLOS topology
+#### LEAF Roles for CLOSE topology
```
Comment (blackliner) [inline docs/release-notes/index.md:70]: ok this one is unwanted

File: docs/vlab/running.md [hunk 4]
```diff
@@ -8,10 +8,10 @@ before running VLAB.
 As a first step you need to initialize Fabricator for the VLAB by running `hhfab init --preset vlab` (or `-p vlab`). It
 supports a lot of customization options which you can find by adding `--help` to the command. If you want to tune the
 topology used for the VLAB you can use `--fabric-mode` (or `-m`) flag to choose between `spine-leaf` (default) and
-`collapsed-core` topologies as well as you can configure the number of spines, leafs, connections and etc. For example,
-`--spines-count` and `--mclag-leafs-count` flags allows to set number of spines and MCLAG leafs respectively.
+`collapsed-core` topologies as well as you can configure the number of spines, leaves, connections and etc. For example,
+`--spines-count` and `--mclag-leaves-count` flags allows to set number of spines and MCLAG leaves respectively.
```
Comment (blackliner) [inline docs/vlab/running.md:12]: this could be problematic, if program arguments use `leafs` instead of `leaves`

File: docs/vlab/demo.md (hunk 2 above)
Comment (Frostman) [reply docs/vlab/demo.md:10]: As discussed in Slack, I think it'll be easier and more consistent to allow "leafs" for now in the typo checker as CLI flags & output already contain a lot of it.

**Review by Frostman — CHANGES_REQUESTED** [review]

File: mkdocs.yml (hunk 1 above)
Comment (blackliner) [reply mkdocs.yml]: fixed, using google ymlfmt now

File: docs/vlab/demo.md (hunk 2 above)
Comment (blackliner) [reply docs/vlab/demo.md]: added leafs as an extra word in the typos config

File: docs/vlab/running.md (hunk 4 above)
Comment (blackliner) [reply docs/vlab/running.md]: done

File: docs/architecture/fabric.md [hunk 5]
```diff
@@ -26,20 +26,20 @@ simple and is ideal for very small deployments.
...
 ### Spine - Leaf
 
-A spine-leaf topology is a standard clos network with workloads attaching to leaf switches and spines providing
+A spine-leaf topology is a standard close network with workloads attaching to leaf switches and spines providing
 connectivity between different leaves.
 
 ![image](./fabric-spineleaf.png)
 
-This kind of topology is useful for bigger deployments and provides all the advantages of a typical clos network.
+This kind of topology is useful for bigger deployments and provides all the advantages of a typical close network.
```
[the remark this answers is not in the document]
Comment (blackliner) [reply docs/architecture/fabric.md:34]: done

File: docs/install-upgrade/build-wiring.md [hunk 6]
```diff
@@ -25,10 +25,10 @@ OPTIONS:
    --chain-control-link         chain control links instead of all switches directly connected to control node if fabric mode is spine-leaf (default: false)
    --control-links-count value  number of control links if chain-control-link is enabled (default: 0)
    --fabric-links-count value   number of fabric links if fabric mode is spine-leaf (default: 0)
-   --mclag-leafs-count value    number of mclag leafs (should be even) (default: 0)
+   --mclag-leaves-count value    number of mclag leaves (should be even) (default: 0)
```
[the remark this answers is not in the document]
Comment (blackliner) [reply docs/install-upgrade/build-wiring.md]: done

File: docs/install-upgrade/build-wiring.md [hunk 7]
```diff
@@ -25,10 +25,10 @@ OPTIONS:
...
    --mclag-peer-links value     number of mclag peer links for each mclag leaf (default: 0)
    --mclag-session-links value  number of mclag session links for each mclag leaf (default: 0)
-   --orphan-leafs-count value   number of orphan leafs (default: 0)
+   --orphan-leaves-count value   number of orphan leaves (default: 0)
```
[the remark this answers is not in the document]
Comment (blackliner) [reply docs/install-upgrade/build-wiring.md]: done

File: docs/reference/api.md [hunk 8]
```diff
@@ -90,7 +90,7 @@ _Appears in:_
 
 ## dhcp.githedgehog.com/v1alpha2
 
-Package v1alpha2 contains API Schema definitions for the dhcp v1alpha2 API group. It is the primarely internal API
+Package v1alpha2 contains API Schema definitions for the dhcp v1alpha2 API group. It is the primarily internal API
```
[the remark this answers is not in the document]
Comment (blackliner) [reply docs/reference/api.md:93]: added to exclude list

File: docs/vlab/running.md [hunk 9]
```diff
@@ -20,7 +20,7 @@ ubuntu@docs:~$ hhfab init -p vlab
 01:17:44 INF Generating wiring from gen flags
 01:17:44 INF Building wiring diagram fabricMode=spine-leaf chainControlLink=false controlLinksCount=0
 01:17:44 INF                     >>> spinesCount=2 fabricLinksCount=2
-01:17:44 INF                     >>> mclagLeafsCount=2 orphanLeafsCount=1
+01:17:44 INF                     >>> mclagLeavesCount=2 orphanLeavesCount=1
```
[the remark this answers is not in the document]
Comment (blackliner) [reply docs/vlab/running.md]: done

File: docs/vlab/running.md [hunk 10]
```diff
@@ -33,21 +33,21 @@ Or if you want to run Collapsed Core topology with 2 MCLAG switches:
 ubuntu@docs:~$ hhfab init -p vlab -m collapsed-core
 01:20:07 INF Generating wiring from gen flags
 01:20:07 INF Building wiring diagram fabricMode=collapsed-core chainControlLink=false controlLinksCount=0
-01:20:07 INF                     >>> mclagLeafsCount=2 orphanLeafsCount=0
+01:20:07 INF                     >>> mclagLeavesCount=2 orphanLeavesCount=0
```
[the remark this answers is not in the document]
Comment (blackliner) [reply docs/vlab/running.md]: done

File: docs/vlab/running.md [hunk 11]
```diff
@@ -33,21 +33,21 @@ Or if you want to run Collapsed Core topology with 2 MCLAG switches:
...
+Or you can run custom topology with 2 spines, 4 MCLAG leaves and 2 non-MCLAG leaves using flags:
 
 ```bash
-ubuntu@docs:~$ hhfab init -p vlab --mclag-leafs-count 4 --orphan-leafs-count 2
+ubuntu@docs:~$ hhfab init -p vlab --mclag-leaves-count 4 --orphan-leaves-count 2
 01:21:53 INF Generating wiring from gen flags
 01:21:53 INF Building wiring diagram fabricMode=spine-leaf chainControlLink=false controlLinksCount=0
 01:21:53 INF                     >>> spinesCount=2 fabricLinksCount=2
-01:21:53 INF                     >>> mclagLeafsCount=4 orphanLeafsCount=2
+01:21:53 INF                     >>> mclagLeavesCount=4 orphanLeavesCount=2
```
[the remark this answers is not in the document]
Comment (blackliner) [reply docs/vlab/running.md]: done

File: mkdocs.yml [hunk 12]
```diff
@@ -103,8 +103,8 @@ markdown_extensions:
   - pymdownx.mark
   - pymdownx.tilde
   - pymdownx.emoji:
-      emoji_index: !!python/name:material.extensions.emoji.twemoji
-      emoji_generator: !!python/name:material.extensions.emoji.to_svg
+      emoji_index: !%21python/name:material.extensions.emoji.twemoji
```
[the remark this answers is not in the document]
Comment (blackliner) [reply mkdocs.yml]: done

File: docs/architecture/fabric.md [hunk 13]
```diff
@@ -39,7 +39,7 @@ underlay network.
 ## Overlay Network
 
 The overlay network runs on top the underlay network to create a virtual network. The overlay network isolates control
-and data plane traffic between different virtual networks and the underlay network. Vitualization is achieved in the
+and data plane traffic between different virtual networks and the underlay network. Visualization is achieved in the
```
Comment (sonoble) [inline docs/architecture/fabric.md:42]: Visualization?

Comment (Frostman) [reply docs/architecture/fabric.md:42]: Not sure what was intended here :) I will fix it separately.

**Review by Frostman — APPROVED** [review]

**Pull request merged**